### PR TITLE
feat: create span question from SDK

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR introduces support for configuring feedback datasets with span questions.  A new `SpanQuestion` class is defined for that. The creation workflow would be as follows:

```python
dataset = rg.FeedbackDataset(
   fields = [rg.TextField(name="text"),...],
   questions=[rg.SpanQuestion(name="spans", title="This is the Span question", labels=["person", "org"])],
)

## You can also create label options with a different text that will shown in the annotation UI
rg.SpanQuestion(
   name="spans", 
   title="This is the Span question", 
   labels=[
      rg.SpanLabelOption(value="person", text="Person"), 
      rg.SpanLabelOption(
         value="org", 
         text="Organization", 
         description="This is a description for label org which could be useful as contextual help from the UI"
      )
   ]
)
```

Closes https://github.com/argilla-io/argilla/issues/4616

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

Tested locally using a running instance of the Argilla server.

**Checklist**

- [ ] I added relevant documentation
- [X] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by frascuchon** [comment]
@nataliaElv I've included you as a reviewer to validate the creation workflow. Let me know if you miss something.

@burtenshaw I've included you to look at the development flow for the current SDK.

File: src/argilla/client/feedback/schemas/questions.py [hunk 1]
```diff
@@ -296,3 +298,65 @@ def server_settings(self) -> Dict[str, Any]:
...
+        value: The value of the span label. This is the value that will be shown in the UI.
+        text: The text of the span label. This is the text that will be shown in the UI.
+
+    Examples:
+        >>> from argilla.client.feedback.schemas.questions import SpanLabelOption
+        >>> SpanLabelOption(value="span_label_1", text="Span Label 1")
+    """
+
+    value: str
+    text: Optional[str]
```
Comment (jfcalvo) [inline src/argilla/client/feedback/schemas/questions.py:317]: I'm missing `description` as optional attribute here.

**Review by jfcalvo — CHANGES_REQUESTED** [review]

File: src/argilla/client/feedback/schemas/questions.py (hunk 1 above)
Comment (frascuchon) [reply src/argilla/client/feedback/schemas/questions.py:317]: done

**Review by jfcalvo — APPROVED** [review]

**Comment by jfcalvo** [comment]
Looks like there is a failing integration test related with span and a dataset deletion (probably because the dataset was not created well?)

**Pull request merged**